src/Provider/Pelias/Tests/PeliasTest.php

```diff
@@ -0,0 +1,143 @@
+        $this->assertEquals(0, $result->count());
+    }
+
+    public function testReverse()
+    {
+        $provider = new Pelias($this->getMockedHttpClient('{}'), 'http://localhost/');
+        $result = $provider->reverseQuery(ReverseQuery::fromCoordinates(0, 0));
+
+        $this->assertInstanceOf(Collection::class, $result);
+        $this->assertEquals(0, $result->count());
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\QuotaExceeded
+     * @expectedExceptionMessage Valid request but quota exceeded.
+     */
+    public function testGeocodeQuotaExceeded()
+    {
+        $provider = new Pelias(
+            $this->getMockedHttpClient(
+                '{
+                    "meta": {
+                        "version": 1,
+                        "status_code": 429
+                    },
+                    "results": {
+                        "error": {
+                            "type": "QpsExceededError",
+                            "message": "Queries per second exceeded: Queries exceeded (6 allowed)."
+                        }
+                    }
+                }'
+            ),
+            'http://localhost/'
+        );
+        $provider->geocodeQuery(GeocodeQuery::create('New York'));
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\InvalidCredentials
+     * @expectedExceptionMessage Invalid or missing api key.
+     */
+    public function testGeocodeInvalidApiKey()
+    {
+        $provider = new Pelias(
+            $this->getMockedHttpClient(
+                '{
+                    "meta": {
+                        "version": 1,
+                        "status_code": 403
+                    },
+                    "results": {
+                        "error": {
+                            "type": "KeyError",
+                            "message": "No api_key specified."
+                        }
+                    }
+                }'
+            ),
+            'http://localhost/'
+        );
+        $provider->geocodeQuery(GeocodeQuery::create('New York'));
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\UnsupportedOperation
+     * @expectedExceptionMessage The Pelias provider does not support IP addresses, only street addresses.
+     */
+    public function testGeocodeWithLocalhostIPv4()
+    {
+        $provider = new Pelias($this->getMockedHttpClient(), 'http://localhost/');
+        $provider->geocodeQuery(GeocodeQuery::create('127.0.0.1'));
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\UnsupportedOperation
+     * @expectedExceptionMessage The Pelias provider does not support IP addresses, only street addresses.
+     */
+    public function testGeocodeWithLocalhostIPv6()
+    {
+        $provider = new Pelias($this->getMockedHttpClient(), 'http://localhost/');
+        $provider->geocodeQuery(GeocodeQuery::create('::1'));
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\UnsupportedOperation
```

Comment:
I think `expectedException` is remove in the latest version of phpunit.
Maybe worth using `$this->expectedException()` instead?

Reply:
That's a copy-paste from the other providers.
But you, indeed, seem to be right!